File: components/mdz/internal/model/organization.go
package model

import "time"

type Organization struct {
	LegalName            string    `json:"legalName"`
	ParentOrganizationID *string   `json:"parentOrganizationId,omitempty"`
	DoingBusinessAs      string    `json:"doingBusinessAs"`
	LegalDocument        string    `json:"legalDocument"`
	Status               Status    `json:"status"`
	Address              Address   `json:"address"`
	Metadata             *Metadata `json:"metadata,omitempty"`
}

type Status struct {
	Code        *string `json:"code,omitempty"`
	Description string  `json:"description"`
}

type Address struct {
	Line1   *string `json:"line1,omitempty"`
	Line2   *string `json:"line2,omitempty"`
	ZipCode *string `json:"zipCode,omitempty"`
	City    *string `json:"city,omitempty"`
	State   *string `json:"state,omitempty"`
	Country string  `json:"country"`
}

type Metadata struct {
	Chave   *string  `json:"chave,omitempty"`
	Bitcoin *string  `json:"bitcoinn,omitempty"`
	Boolean *bool    `json:"boolean,omitempty"`
	Double  *float64 `json:"double,omitempty"`
	Int     *int     `json:"int"`
}

type OrganizationCreate struct {
	ID                   string   `json:"id"`
	ParentOrganizationID string   `json:"parentOrganizationId"`
	LegalName            string   `json:"legalName"`
	DoingBusinessAs      string   `json:"doingBusinessAs"`
	LegalDocument        string   `json:"legalDocument"`
	Address              Address  `json:"address"`
	Status               Status   `json:"status"`
	CreatedAt            string   `json:"createdAt"`
	UpdatedAt            string   `json:"updatedAt"`
	DeletedAt            string   `json:"deletedAt"`
	Metadata             Metadata `json:"metadata"`
}

type OrganizationList struct {
	Items []OrganizationItem `json:"items"`
	Page  int                `json:"page"`
	Limit int                `json:"limit"`
}

type OrganizationItem struct {
	ID                   string     `json:"id"`
	ParentOrganizationID *string    `json:"parentOrganizationId"`
	LegalName            string     `json:"legalName"`
	DoingBusinessAs      string     `json:"doingBusinessAs"`
	LegalDocument        string     `json:"legalDocument"`
	Address              Address    `json:"address"`
	Status               Status     `json:"status"`
	CreatedAt            time.Time  `json:"createdAt"`
	UpdatedAt            time.Time  `json:"updatedAt"`
	DeletedAt            *time.Time `json:"deletedAt"`
	Metadata             *Metadata  `json:"metadata"`
}

type OrganizationUpdate struct {
	ParentOrganizationID *string       `json:"parentOrganizationId,omitempty"`
	LegalName            string        `json:"legalName"`
	DoingBusinessAs      string        `json:"doingBusinessAs"`
	LegalDocument        *string       `json:"legalDocument,omitempty"`
	Address              Address       `json:"address"`
	Status               *StatusUpdate `json:"status,omitempty"`
	Metadata             *Metadata     `json:"metadata,omitempty"`
}

type StatusUpdate struct {
	Code        *string `json:"code,omitempty"`
	Description *string `json:"description,omitempty"`
}

File: components/console/src/schema/organization.ts
import { z } from 'zod'
import { address } from './address'
import { onlyNumbers, regex } from './regex'
import { metadata } from './metadata'

const id = z.string().optional()

const parentOrganizationId = z.string().optional().nullable()

const legalName = z.string().min(1).max(255)

const doingBusinessAs = z.string().min(1).max(100).optional()

const legalDocument = z
  .string()
  .min(1)
  .max(255)
  .refine(regex(onlyNumbers), {
    params: { id: 'custom_only_numbers' }
  })

const accentColor = z.string().optional()

const avatar = z.string().optional()

const status = z
  .object({
    code: z.string(),
    description: z.string()
  })
  .default({
    code: 'ACTIVE',
    description: 'organization is active'
  })

export const organization = {
  id,
  parentOrganizationId,
  legalName,
  doingBusinessAs,
  legalDocument,
  metadata,
  accentColor,
  avatar,
  status,
  address
}
